src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -485,23 +484,53 @@ private void deleteKeyFile() {
     private synchronized KeyPair getKeyPairFromAndroidKeyStore() throws GeneralSecurityException, IOException {
         KeyStore keyStore = KeyStore.getInstance("AndroidKeyStore");
         keyStore.load(null);
-        if (!keyStore.containsAlias(KEY_STORE_CERT_ALIAS)) {
-            Logger.v(TAG, "Key entry is not available");
-            Calendar start = Calendar.getInstance();
-            Calendar end = Calendar.getInstance();
-            end.add(Calendar.YEAR, 100);
-
-            // self signed cert stored in AndroidKeyStore to asym. encrypt key
-            // to a file
-            final KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA",
-                    "AndroidKeyStore");
-            generator.initialize(getKeyPairGeneratorSpec(mContext, start.getTime(), end.getTime()));
-            generator.generateKeyPair();
-            Logger.v(TAG, "Key entry is generated");
-        } else {
-            Logger.v(TAG, "Key entry is available");
+        
+        try {
+            if (!keyStore.containsAlias(KEY_STORE_CERT_ALIAS)) {
+                Logger.v(TAG, "Key entry is not available");
+                Calendar start = Calendar.getInstance();
```

Comment:
Since NullPointerException is such a broad and dangerous exception I'd like to catch it only around the functions that we think might throw it due to a bug.  I think it might be better to call keyStore.containsAlias with in its own try catch, save the result and then use it in an 'if' statement.  The you can have a second try{}catch(NullPointerException) wrapped only around the keystore methods.  You would then be able to exclude the date and logging functions from the try catch.